src/ray/rpc/gcs_server/gcs_rpc_server.h

```diff
@@ -141,7 +141,7 @@ namespace rpc {
 
 // TODO(vitsai): Set auth for everything except GCS.
 #define INTERNAL_KV_SERVICE_RPC_HANDLER(HANDLER) \
-  RPC_SERVICE_HANDLER(InternalKVGcsService, HANDLER, -1)
+  RPC_SERVICE_HANDLER_CUSTOM_AUTH(InternalKVGcsService, HANDLER, -1, AuthType::NO_AUTH)
```

Comment:
Internal KV should also have auth?

Reply:
done, adding it back if there is an unexpected test failure